src/backend/distributed/commands/distribute_object_ops.c

```diff
@@ -337,6 +337,41 @@ static DistributeObjectOps Procedure_Rename = {
 	.postprocess = NULL,
 	.address = RenameFunctionStmtObjectAddress,
 };
+static DistributeObjectOps Routine_AlterObjectDepends = {
+	.deparse = (void *) DeparseAlterFunctionDependsStmt,
+	.qualify = (void *) QualifyAlterFunctionDependsStmt,
+	.preprocess = (void *) PreprocessAlterFunctionDependsStmt,
+	.postprocess = NULL,
+	.address = (void *) AlterFunctionDependsStmtObjectAddress,
+};
+static DistributeObjectOps Routine_AlterObjectSchema = {
+	.deparse = (void *) DeparseAlterFunctionSchemaStmt,
+	.qualify = (void *) QualifyAlterFunctionSchemaStmt,
+	.preprocess = (void *) PreprocessAlterFunctionSchemaStmt,
+	.postprocess = (void *) PostprocessAlterFunctionSchemaStmt,
+	.address = (void *) AlterFunctionSchemaStmtObjectAddress,
+};
+static DistributeObjectOps Routine_AlterOwner = {
+	.deparse = (void *) DeparseAlterFunctionOwnerStmt,
+	.qualify = (void *) QualifyAlterFunctionOwnerStmt,
+	.preprocess = (void *) PreprocessAlterFunctionOwnerStmt,
+	.postprocess = NULL,
+	.address = (void *) AlterFunctionOwnerObjectAddress,
+};
+static DistributeObjectOps Routine_Drop = {
+	.deparse = (void *) DeparseDropFunctionStmt,
+	.qualify = NULL,
```

Comment:
are routines not schema scoped?

**edit:** ah, never mind, drop statements could have multiple references in there for which not all are distributed :). We implement qualification now in the deparsing. Might want to revisit in the future but ok for now.

**edit2:** reading more of this patch and our code, I guess it will be hard to solve due to the `"any"` syntax case :(.